src/common/snippets/docs/snippets_design_guide.md

```diff
@@ -1,301 +1,465 @@
-# SnippetS Design Guide
-This document describes the design and rationale for a snippets code generator. Implementation of code functionality is located [here](https://github.com/openvinotoolkit/openvino/tree/master/src/common/snippets). A proposal for CPU backend integration is [here](https://github.com/openvinotoolkit/openvino/pull/2824).
+# Snippets Design Guide
+## Introduction
 
-## Rationale
+Any artificial neural network topology can be represented as a sequence of mathematical operations (or `Ops`) on some data. The first thing we need to do in order to evaluate an `Op` is to load its input data to the processing unit (typically CPU or GPU, we consider CPU below for definiteness). Then the unit can perform all the necessary manipulations with the data. When this stage is finished, the computation results should be stored back to memory to make space for the next portion of input data, and also to make the results available to subsequent `Ops`. So basically an `Op` execution time is defined by two factors: how fast we can move data from the memory to the processor (and back), and how fast the processor can perform the computations. These factors are called memory and computation bandwidth, respectively.
```

Comment:
It'd be really great if you split the sentences by lines. Markdown thinks that it's new line only if there is empty line between paragraph. It's easier to review and suggest changes